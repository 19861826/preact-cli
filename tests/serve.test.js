import { resolve } from 'path';
import mkdirp from 'mkdirp';
import rimraf from 'rimraf';
import promisify from 'es6-promisify';
import uuid from 'uuid/v4';
import { By } from 'selenium-webdriver';
import run, { spawn } from './run';
import createDriver from './selenium/driver';

const rm = promisify(rimraf);

describe('preact serve', () => {
	let workDir = '';
	let serve;
	let driver;

	beforeEach(async () => {
		workDir = resolve(__dirname, 'output', uuid());
		await mkdirp(workDir);
		driver = await createDriver();
	});

	afterEach(async () => {
		if (serve) {
			serve.kill();
		}

		await rm(workDir);
		await driver.quit();
	});

	it(`should create server hosting the app.`, async () => {
		await run(['create', 'app', '--no-install'], workDir);
		let appDir = resolve(workDir, 'app');
		await run(['build'], appDir);
		serve = spawn(['serve'], appDir);

		// Wait for server to start
		await driver.sleep(500);
		await driver.get('https://localhost:8080/');

		let html = await driver.findElement(By.css('body'))
			.getAttribute('outerHTML');

		expect(html).toMatchSnapshot();
	});
});
