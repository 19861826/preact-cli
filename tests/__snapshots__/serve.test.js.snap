// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preact serve should create server hosting the app. 1`] = `"<body><div id=\\"app\\"><header class=\\"header__3fP58\\"><h1>Preact App</h1><nav><a href=\\"/\\" class=\\"active__2aRKV\\">Home</a><a href=\\"/profile\\" class=\\"\\">Me</a><a href=\\"/profile/john\\" class=\\"\\">John</a></nav></header><div class=\\"home__MVGbg\\"><h1>Home</h1><p>This is the Home component.</p></div></div><script src=\\"/bundle.js\\" async=\\"\\"></script></body>"`;
